maven-embedder/src/main/java/org/apache/maven/cli/logging/Slf4jLoggerManager.java

```diff
@@ -36,11 +36,16 @@
     implements LoggerManager
 {
 
-    private ILoggerFactory loggerFactory;
+    private final ILoggerFactory loggerFactory;
 
     public Slf4jLoggerManager()
     {
-        loggerFactory = LoggerFactory.getILoggerFactory();
+        this( LoggerFactory.getILoggerFactory() );
+    }
+
+    public Slf4jLoggerManager( ILoggerFactory loggerFactory )
```

Comment:
When would this be useful?

Reply:
Do we need an `@since` here?

Reply:
It would make sense to use it in `MavenCli` as we already have the logger factory:
  https://github.com/apache/maven/blob/e019bc93abc9525136e8df7cf897645f29010c3f/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java#L605
I think I had this initially, but I can remove the additional constructor.

Reply:
Go ahead. Lese code is better.

Reply:
I removed the unused constructor.